File: web/JS/stockPage.js

var str = document.getElementById('searchContent').value; //String to store the searched symbol for the API
console.log(document.getElementById('searchContent').value);

var str1; //String used to store the company name in the heading of the chart

window.onload = function () {
    
    var dps1 = [], dps2= [];
    var stockChart = new CanvasJS.StockChart("chartContainer",{
      theme: "light2",
      exportEnabled: true,
      title:{
        text: str1
      },
      subtitles: [{
        text: str +" Price (in USD)"
      }],
      charts: [{
        axisX: {
          crosshair: {
            enabled: true,
            snapToDataPoint: true
          }
        },
        axisY: {
          prefix: "$"
        },
        data: [{
          type: "line",
          yValueFormatString: "$#,###.##",
          dataPoints : dps2
        }]
      }]
     
    });

    

      //API Call to get the chart data
      const yfinance = new XMLHttpRequest();
      const url = 'https://yfapi.net/v8/finance/chart/'+str+'?range=1d&region=US&interval=1m&lang=en';

      yfinance.open('GET', url, true);
      yfinance.setRequestHeader('accept', 'application/json');
      yfinance.setRequestHeader('x-api-key', 'JqFd3HDR5z6iR67p6O9PDLjrvAZtexr255MsgBKa');
      

      yfinance.onload = function () {
        const d = JSON.parse(this.response);
        
        arr = d.chart.result[0].indicators.quote[0].close;
        time = d.chart.result[0].timestamp;
               
        
        for (let i = 0; i < time.length; i++) {
          dps2.push({x:new Date(time[i]*1000), y: arr[i]});

                
        }
      stockChart.render();

        
      };
      yfinance.send();




//API function to get the price chart details of the day/month/year
function graph(cal,url){
  document.getElementById(cal).onclick = function(){
    var dps1 = [], dps2= [];
    var stockChart = new CanvasJS.StockChart("chartContainer",{
      theme: "light2",
      exportEnabled: true,
      title:{
        text: str1
      },
      subtitles: [{
        text: str +" Price (in USD)"
      }],
      charts: [{
        axisX: {
          crosshair: {
            enabled: true,
            snapToDataPoint: true
          }
        },
        axisY: {
          prefix: "$"
        },
        data: [{
          type: "line",
          yValueFormatString: "$#,###.##",
          dataPoints : dps2
        }]
      }]
      
    });

      const yfinance = new XMLHttpRequest();
      
      yfinance.open('GET', url, true);
      yfinance.setRequestHeader('accept', 'application/json');
      yfinance.setRequestHeader('x-api-key', 'JqFd3HDR5z6iR67p6O9PDLjrvAZtexr255MsgBKa');
      

      yfinance.onload = function () {
        const d = JSON.parse(this.response);
        
        arr = d.chart.result[0].indicators.quote[0].close;
        time = d.chart.result[0].timestamp;
        console.log('graph');
        
        
        for (let i = 0; i < time.length; i++) {
          dps2.push({x:new Date(time[i]*1000), y: arr[i]});

                
        }
      stockChart.render();

        
      };
      yfinance.send();
    };
}


graph('day','https://yfapi.net/v8/finance/chart/'+str+'?range=1d&region=US&interval=1m&lang=en');
graph('month','https://yfapi.net/v8/finance/chart/'+str+'?range=1y&region=US&interval=1d&lang=en');
graph('year','https://yfapi.net/v8/finance/chart/'+str+'?range=max&region=US&interval=1d&lang=en');



};





//Function to fill the company name, price, and such details which get refreshed every two seconds 

function yfinanc() {
    var yfinance = new XMLHttpRequest();
   
    url = 'https://yfapi.net/v6/finance/quote?region=US&lang=en&symbols='+str;
    console.log(url);
    yfinance.open('GET', url, true);
    yfinance.setRequestHeader('accept', 'application/json');
    yfinance.setRequestHeader('x-api-key', 'JqFd3HDR5z6iR67p6O9PDLjrvAZtexr255MsgBKa');
    
    yfinance.onload = function () {
        // Begin accessing JSON data here
        var d = JSON.parse(this.response);
                
        var compName = d.quoteResponse.result[0].longName;
        str1=compName;
        var sym = d.quoteResponse.result[0].symbol;
        var price = d.quoteResponse.result[0].regularMarketPrice;
        
        var mchange=d.quoteResponse.result[0].regularMarketChange;
        mchange = mchange.toFixed(2);

        var pchange = d.quoteResponse.result[0].regularMarketChangePercent;
        pchange = pchange.toFixed(3);
        
        
        document.getElementById("compName").innerHTML = compName;
        document.getElementById("title").innerHTML = compName;
        document.getElementById("symbol").innerHTML = sym;
        document.getElementById("price").innerHTML = price ;
        document.getElementById("pchange").innerHTML = mchange + " ("+pchange+"%)" ;

    };
    
     yfinance.send(); 
}

setInterval(yfinanc,2000);

//Function to control the tabs in HTML
function openCity(evt, cityName) {
  var i, tabcontent, tablinks;
  tabcontent = document.getElementsByClassName("tabcontent");
  for (i = 0; i < tabcontent.length; i++) {
    tabcontent[i].style.display = "none";
  }
  tablinks = document.getElementsByClassName("tablinks");
  for (i = 0; i < tablinks.length; i++) {
    tablinks[i].className = tablinks[i].className.replace(" active", "");
  }
  document.getElementById(cityName).style.display = "block";
  evt.currentTarget.className += " active";
}




//Function to get the stock summary data using the API and fill in the html fields
function stock_summary(){
  url = 'https://yfapi.net/v6/finance/quote?region=US&lang=en&symbols='+str;

  var summary = new XMLHttpRequest();
  summary.open('GET', url, true);
  summary.setRequestHeader('accept', 'application/json');
  summary.setRequestHeader('x-api-key', 'JqFd3HDR5z6iR67p6O9PDLjrvAZtexr255MsgBKa');
  summary.onload = function(){
    var d = JSON.parse(this.response);
    

    var regularMarketPreviousClose = d.quoteResponse.result[0].regularMarketPreviousClose;
    var regularMarketOpen  = d.quoteResponse.result[0].regularMarketOpen;
    var regularMarketDayRange = d.quoteResponse.result[0].regularMarketDayRange;
    var regularMarketChange = d.quoteResponse.result[0].regularMarketChange;
    var ask = d.quoteResponse.result[0].ask;
    var bid = d.quoteResponse.result[0].bid;
    var fiftyTwoWeekRange = d.quoteResponse.result[0].fiftyTwoWeekRange;
    var fiftyDayAverageChange = d.quoteResponse.result[0].fiftyDayAverageChange;
    var twoHundredDayAverage = d.quoteResponse.result[0].twoHundredDayAverage;
    var twoHundredDayAverageChange = d.quoteResponse.result[0].twoHundredDayAverageChange;
    var epsCurrentYear = d.quoteResponse.result[0].epsCurrentYear;
    var epsTrailingTwelveMonths = d.quoteResponse.result[0].epsTrailingTwelveMonths;

    document.getElementById('regularMarketPreviousClose').innerHTML = regularMarketPreviousClose;
    document.getElementById('regularMarketOpen').innerHTML = regularMarketOpen;
    document.getElementById('regularMarketDayRange').innerHTML = regularMarketDayRange ;
    document.getElementById('regularMarketChange').innerHTML = regularMarketChange;
    document.getElementById('ask').innerHTML = ask;
    document.getElementById('bid').innerHTML = bid;
    document.getElementById('fiftyTwoWeekRange').innerHTML = fiftyTwoWeekRange;
    document.getElementById('fiftyDayAverageChange').innerHTML = fiftyDayAverageChange;
    document.getElementById('twoHundredDayAverage').innerHTML = twoHundredDayAverage;
    document.getElementById('twoHundredDayAverageChange').innerHTML = twoHundredDayAverageChange;
    document.getElementById('epsCurrentYear').innerHTML = epsCurrentYear;
    document.getElementById('epsTrailingTwelveMonths').innerHTML = epsTrailingTwelveMonths;
    
    

  };
  summary.send();
}
stock_summary();

//Function to get the company overview data using the API and fill in the html fields
function company_overview(){
  url = 'https://www.alphavantage.co/query?function=OVERVIEW&symbol='+str+'&apikey=OOEPKQDKWRUGS4HV';

  var comp = new XMLHttpRequest();
  comp.open('GET', url, true);  
  comp.onload = function(){
    var d = JSON.parse(this.response);
    console.log(d.Description);
    document.getElementById('comp_overview').innerHTML = d.Description;
  };
  comp.send();
}
company_overview();


//Function to get the financial analysis data using the API and fill in the html fields
function financial_analysis(){
  url = 'https://yfapi.net/v11/finance/quoteSummary/'+str+'?lang=en&region=US&modules=financialData%20';

  var fin = new XMLHttpRequest();
  fin.open('GET', url, true);
  fin.setRequestHeader('accept', 'application/json');
  fin.setRequestHeader('x-api-key', 'JqFd3HDR5z6iR67p6O9PDLjrvAZtexr255MsgBKa');
  fin.onload = function(){
    var d = JSON.parse(this.response);

    // console.log(d.quoteSummary.result[0].financialData.recommendationKey);

    var recommendationKey = d.quoteSummary.result[0].financialData.recommendationKey;
    var totalCash = d.quoteSummary.result[0].financialData.totalCash.fmt;
    var totalCashPerShare = d.quoteSummary.result[0].financialData.totalCashPerShare.fmt;
    var ebitda = d.quoteSummary.result[0].financialData.ebitda.fmt;
    var totalDebt = d.quoteSummary.result[0].financialData.totalDebt.fmt;
    var quickRatio = d.quoteSummary.result[0].financialData.quickRatio.fmt;
    var currentRatio = d.quoteSummary.result[0].financialData.currentRatio.fmt;
    var totalRevenue = d.quoteSummary.result[0].financialData.totalRevenue.fmt;
    var debtToEquity = d.quoteSummary.result[0].financialData.debtToEquity.fmt;
    var revenuePerShare = d.quoteSummary.result[0].financialData.revenuePerShare.fmt;
    var returnOnAssets = d.quoteSummary.result[0].financialData.returnOnAssets.fmt;
    var returnOnEquity = d.quoteSummary.result[0].financialData.returnOnEquity.fmt;
    var grossProfits = d.quoteSummary.result[0].financialData.grossProfits.fmt;
    var freeCashflow = d.quoteSummary.result[0].financialData.freeCashflow.fmt;
    var operatingCashflow = d.quoteSummary.result[0].financialData.operatingCashflow.fmt;
    var earningsGrowth = d.quoteSummary.result[0].financialData.earningsGrowth.fmt;
    var revenueGrowth = d.quoteSummary.result[0].financialData.revenueGrowth.fmt;
    var grossMargins = d.quoteSummary.result[0].financialData.grossMargins.fmt;
    var ebitdaMargins = d.quoteSummary.result[0].financialData.ebitdaMargins.fmt;
    var operatingMargins = d.quoteSummary.result[0].financialData.operatingMargins.fmt;
    var profitMargins = d.quoteSummary.result[0].financialData.profitMargins.fmt;

    document.getElementById('recommendationKey').innerHTML = recommendationKey;
    document.getElementById('totalCash').innerHTML = totalCash;
    document.getElementById('totalCashPerShare').innerHTML = totalCashPerShare;
    document.getElementById('ebitda').innerHTML = ebitda;
    document.getElementById('totalDebt').innerHTML = totalDebt;
    document.getElementById('quickRatio').innerHTML = quickRatio;
    document.getElementById('currentRatio').innerHTML = currentRatio;
    document.getElementById('totalRevenue').innerHTML = totalRevenue;
    document.getElementById('debtToEquity').innerHTML = debtToEquity;
    document.getElementById('revenuePerShare').innerHTML = revenuePerShare;
    document.getElementById('returnOnAssets').innerHTML = returnOnAssets;
    document.getElementById('returnOnEquity').innerHTML = returnOnEquity;
    document.getElementById('grossProfits').innerHTML = grossProfits;
    document.getElementById('freeCashflow').innerHTML = freeCashflow;
    document.getElementById('operatingCashflow').innerHTML = operatingCashflow;
    document.getElementById('earningsGrowth').innerHTML = earningsGrowth;
    document.getElementById('revenueGrowth').innerHTML = revenueGrowth;
    document.getElementById('grossMargins').innerHTML = grossMargins;
    document.getElementById('ebitdaMargins').innerHTML = ebitdaMargins;
    document.getElementById('operatingMargins').innerHTML = operatingMargins;
    document.getElementById('profitMargins').innerHTML = profitMargins;
    
  };
  fin.send();

}
financial_analysis();

File: src/java/servlets/Stock_page.java

package servlets;

import java.io.IOException;
import java.io.PrintWriter;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;


@WebServlet(name = "Stock_page", urlPatterns = {"/Stock_page"})
public class Stock_page extends HttpServlet {

   /*In this method we get the search result string from the servlet and using that string 
    we call the API in JS and display the following information in html*/
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        
        response.setContentType("text/html;charset=UTF-8");
        String res = request.getParameter("stocks");
        try (PrintWriter out = response.getWriter()) {
            /* TODO output your page here. You may use following sample code. */
            out.println("<!DOCTYPE html>\n" +
                        "<html>\n" +
                        "  <head>\n" +
                        "    <meta charset=\"utf-8\">\n" +
                        "    <meta name=\"viewport\" content=\"width=device-width\">\n" +
                        "    <title id=\"title\">Stock page</title>\n" +
                        "    <link href=\"CSS/stockPage.css\" rel=\"stylesheet\" type=\"text/css\" />\n" +
                        "    <script src=\"https://cdn.jsdelivr.net/npm/chart.js@3.5.1/dist/chart.min.js\"></script>\n" +
                      
                        "      <script src=\"https://canvasjs.com/assets/script/canvasjs.min.js\"></script>\n" +
                        "\n" +
                        "      <script type=\"text/javascript\" src=\"https://canvasjs.com/assets/script/jquery-1.11.1.min.js\"></script>\n" +
                        "      <script type=\"text/javascript\" src=\"https://canvasjs.com/assets/script/canvasjs.stock.min.js\"></script>\n" +
                        "      \n" +
                        "  <body>\n" +
                        "    <div class=\"header\">\n" +
                        "       <h1><img src=\"icons/stockmarket.png\" alt=\"\" height=50 width=50 class=\"icon\"></img> Stock Market</h1>\n" +
                       
                        "    </div>\n" +
                        "<input type=\"hidden\" value=\""+res+"\" class=\"searchContent\" id=\"searchContent\" name=\"searchContent\"/>" +
                        "\n" +
                        "\n" +
                        "<div class=\"row\">\n" +
                        "  <div class=\"col-1 menu\">\n" +
                        "    <ul>\n" +
                        "      <li><form method=\"get\" action=\"Search\">\n"+
//                                + "<img src=\"icons/search.png\" alt=\"\" height=10 width=10></img>" +                                    
//                        "            <input type=\"submit\" value=\"Search\" class=\"start\" ></input>\n"
                                 "<button type=\"submit\" value=\"Search\" class=\"start\" ><img src=\"icons/search.png\" alt=\"\" height=20 width=20 class=\"icon\"></img> Search</button>\n" +
                        "          </form></li>\n" +
                                
                        "      <li><form method=\"get\" action=\"FinancialData\">\n" +         
                        "            <input type=\"hidden\" value=\""+res+"\" class=\"start\" name=\"symbol\"/>\n" +        
                        "            <button type=\"submit\" value=\"Financial Data\" class=\"start\" ><img src=\"icons/Finance.png\" alt=\"\" height=20 width=20 class=\"icon\"></img> Financial Data</button>\n" +
                        "          </form></li>\n" +  
                        
                        "      <li><form method=\"get\" action=\"LivePriceMonitor\">\n" +         
                        "            <button type=\"submit\" value=\"Live Price Monitor\" class=\"start\" ><img src=\"icons/price_monitor.png\" alt=\"\" height=20 width=20 class=\"icon\"></img> Live Price Monitor</button>\n" +
                        "          </form></li>\n" + 
                                
                        
                        "      <li><form method=\"post\" action=\"MyPortfolio\">\n"
                                + "" +         
                        "            <button type=\"submit\" value=\"My Portfolio\" class=\"start\" ><img src=\"icons/Portfolio.png\" alt=\"\" height=20 width=20 class=\"icon\"></img> My Portfolio</button>\n" +
                        "          </form></li>\n" +
                        "    </ul>\n" +
                        "  </div>\n" +
                        "\n" +
                        "  <div class=\"col-9\">\n" +
                        "    <h1 id=\"compName\">Company</h1>\n" +
                        "    <label id=\"symbol\" style=\"padding-left:20px\">symbol</label>\n" +
                        "\n" +
                        "    <div class=\"col-12\">\n" +
                        "      <h2 id=\"price\">price</h2>   \n" +
                        "      <h3 id=\"pchange\">pchange</h3>   \n" +
                        "    </div>\n" +
                        "\n" +
                        "    \n" +
                        "    \n" +
                        "    <div class=\"col-13\">\n" +
                        "      <div id=\"chartContainer\" style=\"height: 500px; width: 100%;\"></div>\n" +
                        "      <table id=\"buttons\" class=\"buttons\">\n" +
                        "      <tr>\n" +
                        "        <button id=\"day\" class=\"chart_buttons\">Days</button>\n" +
                        "        <button id=\"month\" class=\"chart_buttons\">Months</button>\n" +
                        "        <button id=\"year\" class=\"chart_buttons\">Years</button>\n" +
                        "      </tr>\n" +
                        "    </table>\n" +
                        "    </div>\n" +
                        "    \n" +
                        "   \n" +
                        "  \n" +
                        "\n" +
                        "  <div class=\"col-12\">\n" +
                        "    <div class=\"tab\">\n" +
                        "        <button class=\"tablinks\" onclick=\"openCity(event, 'Summary')\">Summary</button>\n" +
                        "        <button class=\"tablinks\" onclick=\"openCity(event, 'Company_Overview')\">Company Overview</button>\n" +
                        "        <button class=\"tablinks\" onclick=\"openCity(event, 'Financial_Analysis')\">Financial Analysis</button>\n" +
                        "        \n" +
                        "    </div>\n" +
                        "\n" +
                        "    <div id=\"Summary\" class=\"tabcontent\">\n" +
                        "          <h3>Summary</h3>\n" +
                        "          <table id=\"Summary_table\">\n" +
                        "            <tr>\n" +
                        "              <td>Previous Close</td>\n" +
                        "              <td id=\"regularMarketPreviousClose\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Open</td>\n" +
                        "              <td id=\"regularMarketOpen\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Day Range</td>\n" +
                        "              <td id=\"regularMarketDayRange\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Market Change</td>\n" +
                        "              <td id=\"regularMarketChange\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Ask</td>\n" +
                        "              <td id=\"ask\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Bid</td>\n" +
                        "              <td id=\"bid\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>52 week range</td>\n" +
                        "              <td id=\"fiftyTwoWeekRange\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>52 week average change</td>\n" +
                        "              <td id=\"fiftyDayAverageChange\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>200 average</td>\n" +
                        "              <td id=\"twoHundredDayAverage\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>200 average change</td>\n" +
                        "              <td id=\"twoHundredDayAverageChange\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>EPS current year</td>\n" +
                        "              <td id=\"epsCurrentYear\">0.00</td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>EPS trailing twelve months</td>\n" +
                        "              <td id=\"epsTrailingTwelveMonths\">0.00</td>\n" +
                        "            </tr>\n" +
                        "          </table>\n" +
                        "          \n" +
                        "    </div>\n" +
                        "    \n" +
                        "    <div id=\"Company_Overview\" class=\"tabcontent\">\n" +
                        "          <h3>Company Overview</h3>\n" +
                        "          <p id=\"comp_overview\">00</p>\n" +
                        "          \n" +
                        "    </div>\n" +
                        "    <div id=\"Financial_Analysis\" class=\"tabcontent\">\n" +
                        "          <h3>Financial Analysis</h3>\n" +
                        "          <table id=\"fin_table\">\n" +
                        "            <tr>\n" +
                        "              <td>Recommendation</td>\n" +
                        "              <td id=\"recommendationKey\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Total Cash</td>\n" +
                        "              <td id=\"totalCash\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Total Cash Per Share</td>\n" +
                        "              <td id=\"totalCashPerShare\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Ebitda</td>\n" +
                        "              <td id=\"ebitda\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Total Debt</td>\n" +
                        "              <td id=\"totalDebt\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Quick Ratio</td>\n" +
                        "              <td id=\"quickRatio\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Current Ratio</td>\n" +
                        "              <td id=\"currentRatio\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Total Revenue</td>\n" +
                        "              <td id=\"totalRevenue\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Debt To Equity</td>\n" +
                        "              <td id=\"debtToEquity\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Revenue Per Share</td>\n" +
                        "              <td id=\"revenuePerShare\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Return On Assets</td>\n" +
                        "              <td id=\"returnOnAssets\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Return On Equity</td>\n" +
                        "              <td id=\"returnOnEquity\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Gross Profits</td>\n" +
                        "              <td id=\"grossProfits\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Free Cash flow</td>\n" +
                        "              <td id=\"freeCashflow\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Operating Cash flow</td>\n" +
                        "              <td id=\"operatingCashflow\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Earnings Growth</td>\n" +
                        "              <td id=\"earningsGrowth\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Revenue Growth</td>\n" +
                        "              <td id=\"revenueGrowth\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Gross Margins</td>\n" +
                        "              <td id=\"grossMargins\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Ebitda Margins</td>\n" +
                        "              <td id=\"ebitdaMargins\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Operating Margins</td>\n" +
                        "              <td id=\"operatingMargins\"></td>\n" +
                        "            </tr>\n" +
                        "            <tr>\n" +
                        "              <td>Profit Margins</td>\n" +
                        "              <td id=\"profitMargins\"></td>\n" +
                        "            </tr>\n" +
                        "            \n" +
                        "          </table>\n" +
                        "          \n" +
                        "    </div>\n" +
                        "\n" +
                        "    \n" +
                        "\n" +
                        "    \n" +
                        "  </div>\n" +
                        "\n" +
                        "\n" +
                        "\n" +
                        "    \n" +
                        "    \n" +
                        "\n" +
                        "   \n" +
                        "</div>\n" +
                        "  </div>\n" +
                                
                                
                        "<div class=\"footer\">\n" +
                        "      <p></p>\n" +
                        "      <h1 style=\"color: #1D2E3E\">Footer</h1>\n" +
                        "    </div>" +
                                
                                
                        "\n" +
                        "\n" +
                        "\n" +
                        "  <script src=\"JS/stockPage.js\"></script>\n" +
                        "  </head>\n" +
                        "  </body>\n" +
                        "</html>");
            
        }
        
    }

  
}
